Userland/Libraries/LibWeb/CSS/CSSStyleDeclaration.cpp

```diff
@@ -209,10 +217,19 @@ void ElementInlineCSSStyleDeclaration::update_style_attribute()
 }
 
 // https://drafts.csswg.org/cssom/#set-a-css-declaration
-bool PropertyOwningCSSStyleDeclaration::set_a_css_declaration(PropertyID property_id, NonnullRefPtr<CSSStyleValue const> value, Important important)
+bool PropertyOwningCSSStyleDeclaration::set_a_css_declaration(PropertyID property_id, StringView property_name, NonnullRefPtr<CSSStyleValue const> value, Important important)
 {
     // FIXME: Handle logical property groups.
 
+    if (property_id == PropertyID::Custom) {
+        auto fly_string_property = FlyString::from_utf8(property_name);
+        if (fly_string_property.is_error()) {
+            return false;
+        }
+        m_custom_properties.set(fly_string_property.release_value(), StyleProperty { .important = important, .property_id = property_id, .value = move(value) });
```

Comment:
`StyleProperty` also has a `custom_name` field you should be setting here. That'll mean not calling `fly_string_property.release_value()` the first time.